### ISSUE TITLE
Focus loss when converting link block into a submenu block

### ISSUE THREAD

**Issue by jeryj** [WordPress/gutenberg#51306]
When adding a submenu link to a custom link, which causes the navigation link block to convert into a submenu block, the focus gets lost and reset to the top of the DOM when the link UI closes. 

### Reproduction Steps
- Open Navigation Sidebar
- On a link that does not have a submenu:
- Click the three dots
- Select Add Submenu Link
- Cancel or select a link
- Focus will be at the top of the DOM

My guess is that it's because the popover Link UI is trying to return focus to the item that opened it, but the item that opened it (the link block) no longer exists, and is now a submenu block.

### Video Reproduction
https://github.com/WordPress/gutenberg/assets/967608/c14ce4b8-d44d-4752-8e87-72569be2863c
